extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -94,7 +104,7 @@ void annotationTransformers(BuildProducer<AnnotationsTransformerBuildItem> trans
                     return;
                 }
                 for (DotName annotationName : ctx.getTarget().asClass().annotations().keySet()) {
-                    if (metricsAnnotations.contains(annotationName)) {
+                    if (DotName.createSimple(Gauge.class.getName()).equals(annotationName)) {
```

Comment:
I'm surprised we change from considering all the annotations to only `Gauge`. What's the reason?

Reply:
We don't need it anymore for other metric types, only gauges are now initialized by the `MetricsInterceptor`, because they can't be initialized eagerly at startup. Other metric types are now initialized by annotation scanning.